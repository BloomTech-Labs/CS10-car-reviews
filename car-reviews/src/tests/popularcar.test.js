import PopularCar from '../components/MainPage/popularcar';

describe('PopularCar', () => {
  const wrapper = shallow(<PopularCar imageURL={['http://tineye.com/images/widgets/mona.jpg']} />);

  // All tests will go here
  it('shallow renders without crashing', () => {
    wrapper;
    expect(wrapper).toMatchSnapshot();
  });

  it('always renders a div', () => {
    const divs = wrapper.find('div');
    expect(divs.length).toBeGreaterThan(0);
  });

  describe('the rendered div', () => {
    it('contains everything else that gets rendered', () => {
      const divs = wrapper.find('div');
      const wrappingDiv = divs.first();

      expect(wrappingDiv.children()).toEqual(wrapper.children());
    });
  });

  describe('Testing state of PopularCar', () => {
    it('searching is defaulted to false', () => {
      expect(wrapper.state('searching')).toBe(false);
    });

    it('searchResults array is empty before searchFunction() called', () => {
      expect(wrapper.state('searchResults')).toHaveLength(0);
    });
  });

  it('Popular Car image should be the image a user uploaded with review', () => {
    const props = {
        imageURL: ['http://tineye.com/images/widgets/mona.jpg']
      },
      PopularCarComponent = mount(<PopularCar {...props} />).find('img');
    expect(PopularCarComponent.prop('src')).toEqual('http://tineye.com/images/widgets/mona.jpg');
  });

  it('Popular Car image should be the default img if user did not upload a car image with review', () => {
    const props = {
        imageURL: []
      },
      PopularCarComponent = mount(<PopularCar {...props} />).find('img');
    expect(PopularCarComponent.prop('src')).toEqual('default_img.png');
  });
});
