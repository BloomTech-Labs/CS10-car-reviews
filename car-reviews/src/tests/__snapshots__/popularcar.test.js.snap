// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularCar shallow renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PopularCar
    imageURL={
        Array [
            "http://tineye.com/images/widgets/mona.jpg",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown />,
        <Button
          className="modal-button"
          color="secondary"
          onClick={[Function]}
          tag="button"
>
          <p
                    style={
                              Object {
                                        "fontSize": "1.1em",
                                      }
                    }
          >
                     undefined undefined
          </p>
          <div
                    style={
                              Object {
                                        "height": "150px",
                                      }
                    }
          >
                    <img
                              alt=""
                              src="http://tineye.com/images/widgets/mona.jpg"
                              style={
                                        Object {
                                                  "height": "100%",
                                                  "maxWidth": "100%",
                                                }
                              }
                    />
          </div>
          <div
                    style={
                              Object {
                                        "display": "flex",
                                        "justifyContent": "center",
                                      }
                    }
          />
</Button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Symbol(react.fragment),
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <p
              style={
                            Object {
                                          "fontSize": "1.1em",
                                        }
              }
>
               undefined undefined
</p>,
            <div
              style={
                            Object {
                                          "height": "150px",
                                        }
              }
>
              <img
                            alt=""
                            src="http://tineye.com/images/widgets/mona.jpg"
                            style={
                                          Object {
                                                        "height": "100%",
                                                        "maxWidth": "100%",
                                                      }
                            }
              />
</div>,
            <div
              style={
                            Object {
                                          "display": "flex",
                                          "justifyContent": "center",
                                        }
              }
/>,
          ],
          "className": "modal-button",
          "color": "secondary",
          "onClick": [Function],
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " undefined undefined",
              "style": Object {
                "fontSize": "1.1em",
              },
            },
            "ref": null,
            "rendered": " undefined undefined",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                src="http://tineye.com/images/widgets/mona.jpg"
                style={
                                Object {
                                                "height": "100%",
                                                "maxWidth": "100%",
                                              }
                }
/>,
              "style": Object {
                "height": "150px",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "src": "http://tineye.com/images/widgets/mona.jpg",
                "style": Object {
                  "height": "100%",
                  "maxWidth": "100%",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "style": Object {
                "display": "flex",
                "justifyContent": "center",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown />,
          <Button
            className="modal-button"
            color="secondary"
            onClick={[Function]}
            tag="button"
>
            <p
                        style={
                                    Object {
                                                "fontSize": "1.1em",
                                              }
                        }
            >
                         undefined undefined
            </p>
            <div
                        style={
                                    Object {
                                                "height": "150px",
                                              }
                        }
            >
                        <img
                                    alt=""
                                    src="http://tineye.com/images/widgets/mona.jpg"
                                    style={
                                                Object {
                                                            "height": "100%",
                                                            "maxWidth": "100%",
                                                          }
                                    }
                        />
            </div>
            <div
                        style={
                                    Object {
                                                "display": "flex",
                                                "justifyContent": "center",
                                              }
                        }
            />
</Button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Symbol(react.fragment),
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <p
                style={
                                Object {
                                                "fontSize": "1.1em",
                                              }
                }
>
                 undefined undefined
</p>,
              <div
                style={
                                Object {
                                                "height": "150px",
                                              }
                }
>
                <img
                                alt=""
                                src="http://tineye.com/images/widgets/mona.jpg"
                                style={
                                                Object {
                                                                "height": "100%",
                                                                "maxWidth": "100%",
                                                              }
                                }
                />
</div>,
              <div
                style={
                                Object {
                                                "display": "flex",
                                                "justifyContent": "center",
                                              }
                }
/>,
            ],
            "className": "modal-button",
            "color": "secondary",
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " undefined undefined",
                "style": Object {
                  "fontSize": "1.1em",
                },
              },
              "ref": null,
              "rendered": " undefined undefined",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  src="http://tineye.com/images/widgets/mona.jpg"
                  style={
                                    Object {
                                                      "height": "100%",
                                                      "maxWidth": "100%",
                                                    }
                  }
/>,
                "style": Object {
                  "height": "150px",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "src": "http://tineye.com/images/widgets/mona.jpg",
                  "style": Object {
                    "height": "100%",
                    "maxWidth": "100%",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "style": Object {
                  "display": "flex",
                  "justifyContent": "center",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
